crates/pathfinder/src/rpc/api.rs

```diff
@@ -162,6 +157,21 @@ impl RpcApi {
             _ => BlockStatus::AcceptedOnL2,
         };
 
+        Ok(block_status)
+    }
+
+    /// This function assumes that the block ID is valid i.e. it won't check if the block hash or number exist.
```

Comment:
An interesting way of handling this with types could be to create a newtype which is verified, and linked by lifetime to the transaction. Bit much for this though :) Was just an interesting thought for me.

Reply:
Interesting idea, I'll have a look